src/mlpack/methods/ann/rbm.hpp

```diff
@@ -0,0 +1,183 @@
+/**
+ * @file rbm.hpp
+ * @author Kris Singh
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_RBM_HPP
+#define MLPACK_METHODS_ANN_RBM_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+#include <mlpack/methods/ann/activation_functions/softplus_function.hpp>
+#include <mlpack/methods/ann/init_rules/gaussian_init.hpp>
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial neural networks.  */ {
+
+template<typename InitializationRuleType, typename RBMPolicy>
+class RBM
+{
+ public:
+  using NetworkType = RBM<InitializationRuleType, RBMPolicy>;
+  typedef typename RBMPolicy::ElemType eT;
```

Comment:
I would go with ``typedef typename VecType::elem_type ElemType;``to be consistent with the rest of the codebase, for more information take a look at: https://github.com/mlpack/mlpack/blob/master/doc/policies/elemtype.hpp